bin/functions/workload-functions.sh

```diff
@@ -448,3 +460,221 @@ INSERT OVERWRITE TABLE uservisits_copy SELECT * FROM uservisits;
 EOF
 
 }
+
+
+function runPowerTest() {
+    DATABASE_NAME="tpcds_${TABLE_SIZE}g"
+
+    SPARK_SQL_CMD="${SPARK_HOME}/bin/spark-sql"
+    SPARK_SQL_GLOBAL_OPTS="--hiveconf hive.metastore.uris=${HIVE_METASTORE_URIS}"
+
+    BEELINE_CMD="${SPARK_HOME}/bin/beeline"
+    BEELINE_GLOBAL_OPTS="-u jdbc:hive2://localhost:10000/${DATABASE_NAME} -n test -p n"
+
+    START_THRIFTSERVER_CMD="${SPARK_HOME}/sbin/start-thriftserver.sh"
+    STOP_THRIFTSERVER_CMD="${SPARK_HOME}/sbin/stop-thriftserver.sh"
+    THRIFTSERVER_GLOBAL_OPTS="--hiveconf hive.metastore.uris=${HIVE_METASTORE_URIS} --conf spark.sql.shuffle.partitions=200"
+
+    INCLUDED_LIST=(19 42 43 52 55 63 68 73 98)
+
+    SET_REDUCE_NUM=()
+    SET_REDUCE_NUM[19]=200
+    SET_REDUCE_NUM[42]=200
+    SET_REDUCE_NUM[43]=200
+    SET_REDUCE_NUM[52]=200
+    SET_REDUCE_NUM[55]=200
+    SET_REDUCE_NUM[63]=200
+    SET_REDUCE_NUM[68]=200
+    SET_REDUCE_NUM[73]=200
+    SET_REDUCE_NUM[98]=200
+
+    export_withlog SPARKBENCH_PROPERTIES_FILES
+
+    QUERY_BEGIN_NUM=${TPCDS_TEST_LIST:0:2}
+    QUERY_END_NUM=${TPCDS_TEST_LIST: -2}
+    len=${#INCLUDED_LIST[@]}
+
+    YARN_OPTS=""
+    if [[ "$SPARK_MASTER" == yarn-* ]]; then
+        export_withlog HADOOP_CONF_DIR
+
+        YARN_OPTS="--num-executors ${YARN_NUM_EXECUTORS}"
+        if [[ -n "${YARN_EXECUTOR_CORES:-}" ]]; then
+            YARN_OPTS="${YARN_OPTS} --executor-cores ${YARN_EXECUTOR_CORES}"
+       fi
+       if [[ -n "${SPARK_YARN_EXECUTOR_MEMORY:-}" ]]; then
+           YARN_OPTS="${YARN_OPTS} --executor-memory ${SPARK_YARN_EXECUTOR_MEMORY}"
+       fi
+       if [[ -n "${SPAKR_YARN_DRIVER_MEMORY:-}" ]]; then
+           YARN_OPTS="${YARN_OPTS} --driver-memory ${SPARK_YARN_DRIVER_MEMORY}"
+       fi
+    fi
+
+    if [ "$TPCDS_SPARKSQLCLI_ENABLED" = "false" ]
+    then
+        ${START_THRIFTSERVER_CMD} --master ${SPARK_MASTER} ${YARN_OPTS} --properties-file ${SPARK_PROP_CONF} ${THRIFTSERVER_GLOBAL_OPTS}
+    fi
+
+    if [ "$TPCDS_SPARKSQLCLI_ENABLED" = "true" ]
+    then
+        echo -e "${BCyan}Running TPC-DS power test with Spark SQL CLI${Color_Off}"
+    else
+        echo -e "${BCyan}Running TPC-DS power test with beeline${Color_Off}"
+    fi
+
+    for (( i=${QUERY_BEGIN_NUM}; i<${QUERY_END_NUM} + 1; i++)); do
```

Comment:
((: i=1,: syntax error: operand expected (error token is ",") when using query list `1,98`

Reply:
Thanks, this has been changed! 
Could you help to review this PR?